x-pack/plugins/lens/public/app_plugin/share_action.ts

```diff
@@ -90,16 +89,19 @@ function getShortShareableURL(
     searchSessionId: data.search.session.getSessionId(),
     references,
     dataViewSpecs: adHocDataViews,
-  });
+  };
 }
 
 export async function getShareURL(
   shortUrlService: (params: LensAppLocatorParams) => Promise<string>,
   services: Pick<LensAppServices, 'application' | 'data'>,
-  configuration: ShareableConfiguration
+  configuration: ShareableConfiguration,
+  shareUrlEnabled: boolean
 ) {
+  const locatorParams = getLocatorParams(services.data, configuration);
   return {
-    shareableUrl: await getShortShareableURL(shortUrlService, services.data, configuration),
+    shareableUrl: await (shareUrlEnabled ? shortUrlService(locatorParams) : undefined),
```

Comment:
I'm seeing this is the code behind the note in the description that a short URL is created every time that a report is requested, but I think we should avoid this.

For Reporting purposes, the `locatorParams` is what is actually used for the `V2` export types: https://github.com/elastic/kibana/blob/1a3ea3a/x-pack/plugins/reporting/public/share_context_menu/register_pdf_png_reporting.tsx#L35-L41, so the short URL is not even used when the report is executing. Unfortunately it is part of the `ShowShareMenuOptions` in the sharing plugin :\

The dashboard app uses `setStateToKbnUrl`: https://github.com/elastic/kibana/blob/503b466/src/plugins/dashboard/public/dashboard_app/top_nav/share/show_share_modal.tsx#L164. Is that an alternative option here?

Reply:
Unfortunately not.
In the Lens case the locator is also registered on the server side to make use of the new shortURL service, which is not possible with the `setStateToKbnUrl` (as it also uses the client storage).
Currently when the user clicks on the `Share` menu a shortURL is already generated, even if the user won't click on the button. This is not great and perhaps can be improved, but it is completely independent from reporting and depends on the fact that the `UrlPanel` wants a static URL string for the creation. I can log a separate issue on this to lazy generate the short URL only when the user clicks on the `Generate` button, but that has an important impact also on the URL share plugin.
Said that, as the short URL is already available, why not use it for reporting?

Reply:
> Said that, as the short URL is already available, why not use it for reporting?

URLs are not used for reporting in the V2 export types. Eventually, the code for V1 types will be removed as usage of those APIs goes down, and all available export types will use locators. 

I agree with you the fact that a shortURL being created is independent of the reporting integration. I also think the Sharing panel should not require a `shareableUrl`, which is another separate issue. cc @vadimkibana

Reply:
Fixed in https://github.com/elastic/kibana/pull/153429/commits/fa61289b5b65c1b5bcfd5e3bca8ccba92671f016

Reply:
Added also a functional test to verify the correctness of the URL generated by reporting in this case ( https://github.com/elastic/kibana/pull/153429/commits/d73df1ef76e3990857f4ce1543a8fc8349441461 )